src/sdk/sdk.ts

```diff
@@ -79,18 +79,25 @@ export async function writeEvent(
 
 /**
  * Get the owner reference for a custom resource
- * @param cr the custom resource to get the owner reference for
- * @returns the owner reference for the custom resource
+ * @param customResource the custom resource to get the owner reference for
+ * @returns the owner reference array for the custom resource
  */
-export function getOwnerRefFrom(cr: GenericKind): V1OwnerReference[] {
-  const { name, uid } = cr.metadata!;
+export function getOwnerRefFrom(
+  customResource: GenericKind,
+  blockOwnerDeletion?: boolean,
+  controller?: boolean,
+): V1OwnerReference[] {
+  const { apiVersion, kind, metadata } = customResource;
+  const { name, uid } = metadata!;
 
   return [
     {
-      apiVersion: cr.apiVersion!,
-      kind: cr.kind!,
+      apiVersion: apiVersion!,
+      kind: kind!,
       uid: uid!,
       name: name!,
+      ...(blockOwnerDeletion !== undefined && { blockOwnerDeletion }),
+      ...(controller !== undefined && { controller }),
```

Comment:
In English, this "Spreads the contents of the corresponding input parameter to the returned object if the parameter was set."